### PR TITLE
DAOS-9994 test: Updating detect() method with missing distros

### PR DESCRIPTION
**Pull request by phender**
Adding support for Rocky Linux to the avocado.utils.distro method.

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

**Comment by phender** [comment]
When using `from distro_utils import detect` in `launch.py` the `testsInStage.grooy` script fails with:
```
[2022-03-01T00:51:54.068Z] + ./list_tests.py pr,-hw
[2022-03-01T00:51:54.068Z] Traceback (most recent call last):
[2022-03-01T00:51:54.068Z]   File "./list_tests.py", line 10, in <module>
[2022-03-01T00:51:54.068Z]     from launch import get_test_list, set_python_environment
[2022-03-01T00:51:54.068Z]   File "/data/jenkins/jenkins-1/light_1/workspace/daos-stack_daos_PR-8342/src/tests/ftest/launch.py", line 34, in <module>
[2022-03-01T00:51:54.068Z]     from distro_utils import detect
[2022-03-01T00:51:54.068Z] ImportError: No module named distro_utils
```

I've switched `launch.py` to use the non-avocado `distro` package as it only needs to know the distro id and major version.  This pacakge does not appear to provide the minor version so I'm still using the new `distro_utils.py` utility in the test to display the distribution information.

Details:
```
$ python3
Python 3.6.8 (default, Nov  9 2021, 14:44:26)
[GCC 8.5.0 20210514 (Red Hat 8.5.0-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import distro
>>> distro.os_release_info()
{'name': 'Rocky Linux', 'version': '8.5 (Green Obsidian)', 'id': 'rocky', 'id_like': 'rhel centos fedora', 'version_id': '8.5', 'platform_id': 'platform:el8', 'pretty_name': 'Rocky Linux 8.5 (Green Obsidian)', 'ansi_color': '0;32', 'cpe_name': 'cpe:/o:rocky:rocky:8:GA', 'home_url': 'https://rockylinux.org/', 'bug_report_url': 'https://bugs.rockylinux.org/', 'rocky_support_product': 'Rocky Linux', 'rocky_support_product_version': '8', 'codename': 'Green Obsidian'}
```
```
$ python3
Python 3.6.8 (default, Sep 21 2021, 20:17:36)
[GCC 8.4.1 20200928 (Red Hat 8.4.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import distro
>>> distro.os_release_info()
{'name': 'CentOS Linux', 'version': '8', 'id': 'centos', 'id_like': 'rhel fedora', 'version_id': '8', 'platform_id': 'platform:el8', 'pretty_name': 'CentOS Linux 8', 'ansi_color': '0;31', 'cpe_name': 'cpe:/o:centos:centos:8', 'home_url': 'https://centos.org/', 'bug_report_url': 'https://bugs.centos.org/', 'centos_mantisbt_project': 'CentOS-8', 'centos_mantisbt_project_version': '8'}
```

**Comment by phender** [comment]
The host that runs list_tests.py does not have the python distro package installed:
```
[2022-03-02T01:08:06.279Z] + ./list_tests.py pr,-hw
[2022-03-02T01:08:06.279Z] Traceback (most recent call last):
[2022-03-02T01:08:06.279Z]   File "./list_tests.py", line 10, in <module>
[2022-03-02T01:08:06.279Z]     from launch import get_test_list, set_python_environment
[2022-03-02T01:08:06.279Z]   File "/data/jenkins/jenkins-1/light_1/workspace/daos-stack_daos_PR-8342/src/tests/ftest/launch.py", line 14, in <module>
[2022-03-02T01:08:06.279Z]     import distro
[2022-03-02T01:08:06.279Z] ImportError: No module named distro
```

**Comment by phender** [comment]
> The host that runs list_tests.py does not have the python distro package installed:
> 
> ```
> [2022-03-02T01:08:06.279Z] + ./list_tests.py pr,-hw
> [2022-03-02T01:08:06.279Z] Traceback (most recent call last):
> [2022-03-02T01:08:06.279Z]   File "./list_tests.py", line 10, in <module>
> [2022-03-02T01:08:06.279Z]     from launch import get_test_list, set_python_environment
> [2022-03-02T01:08:06.279Z]   File "/data/jenkins/jenkins-1/light_1/workspace/daos-stack_daos_PR-8342/src/tests/ftest/launch.py", line 14, in <module>
> [2022-03-02T01:08:06.279Z]     import distro
> [2022-03-02T01:08:06.279Z] ImportError: No module named distro
> ```

Resolved this by going back to using `distro_utils` moving the `from distro_utils import detect` inside the method that uses it.

File: src/tests/ftest/launch.py [hunk 1]
```diff
@@ -1724,6 +1722,8 @@ def install_debuginfos():
         on this node also.
 
     """
+    from distro_utils import detect     # pylint: disable=import-outside-toplevel
```
Comment (daltonbohning) [inline src/tests/ftest/launch.py:1725]: It's also possible to import like this:
```
try:
    from distro_utils import detect
except:
    detect = None
```

File: src/tests/ftest/util/distro_utils.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    CHECK_VERSION_REGEX = re.compile(r"AlmaLinux release (\d{1,2})\.(\d{1,2}).*")
+
+
+distro.register_probe(RockyProbe)
+distro.register_probe(AlmaProbe)
+
+
+def detect():
+    """Detect the current distribution."""
+    return distro.detect()
```
Comment (daltonbohning) [inline src/tests/ftest/util/distro_utils.py:35]: `avocado.utils.distro.detect` doesn't have parameters, but I prefer this approach in the general case so a caller could still pass them. (Could also use `*args` and `**kwargs`)
```suggestion
detect = distro.detect
```

**Review by daltonbohning — APPROVED** [review]
Code-wise, LGTM, though I'm not sure if anything more is required.

File: src/tests/ftest/util/distro_utils.py (hunk 2 above)
Comment (phender) [reply src/tests/ftest/util/distro_utils.py]: Thanks for the suggestion.  I took it a little further.

File: src/tests/ftest/launch.py (hunk 1 above)
Comment (phender) [reply src/tests/ftest/launch.py:1725]: Yes, thanks. This is really temporary until CORCI-1100 is fixed.  We could do this and add better handling of this situation or when `detect()` returns `UNKNOWN_DISTRO_NAME`.  I plan to address issues like this as part of DAOS-9828.

**Review by brianjmurrell — APPROVED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Pull request merged**